<!--
Copyright (c) 2024 Analog Devices, Inc. All Rights Reserved.
This software is proprietary to Analog Devices, Inc. and its licensors.
-->

<!-- -*- mode: XML -*- -->
<launch>

  <!-- Arguments on name space and sensor name in gazebo -->
  <arg name="ns_prefix" default="tmc_coe1" />
  
  <!-- All arguments below belongs to tf frame -->
  <arg name="tmc_coe_base_name" default="tmcm_1461_CoE" />
 
  <!-- "period" specifies how often to send a transform-->
  <arg name="period" default="100"/>
 
  <!-- TMCL Board frame
    change values of:
    * "dev_parent_frame" to specify parent frame of tmcl board (setting to "map" as default)
    * "pos_x,y,z" for translation
    * "pos_qx,qy,qz,qw" for rotation[Quaternion] -->
  <arg name = "dev_parent_frame" value = "map" />
  <arg name = "dev_base_name" value = "$(eval arg('ns_prefix') + '/' + arg('tmc_coe_base_name') + '_base_link')" />
  <arg name = "dev_pos_x" value = "0" />
  <arg name = "dev_pos_y" value = "0" />
  <arg name = "dev_pos_z" value = "0" />
  <arg name = "dev_pos_qx" value = "0" />
  <arg name = "dev_pos_qy" value = "0" />
  <arg name = "dev_pos_qz" value = "0" />
  <arg name = "dev_pos_qw" value = "1" />
 
  <!-- Motor 0 frame
    change values of:
    * "pos_x,y,z" for translation
    * "pos_qx,qy,qz,qw" for rotation[Quaternion] -->
  <arg name = "mtr0_base_name" value = "$(eval arg('ns_prefix') + '/' + arg('tmc_coe_base_name') + '_mtr0_frame')" />
  <arg name = "mtr0_pos_x" value = "1" />
  <arg name = "mtr0_pos_y" value = "0" />
  <arg name = "mtr0_pos_z" value = "0" />
  <arg name = "mtr0_pos_qx" value = "0" />
  <arg name = "mtr0_pos_qy" value = "0" />
  <arg name = "mtr0_pos_qz" value = "0" />
  <arg name = "mtr0_pos_qw" value = "1" />
  
  <group ns="$(arg ns_prefix)">
    <!--Launches node-->
    <node name="tmc_coe_ros_node" pkg="adi_tmc_coe" type="tmc_coe_ros_node" output="screen" required="true" launch-prefix="ethercat_grant">
      <rosparam command="load" file="$(find adi_tmc_coe)/config/autogenerated/TMCM-1461.yaml"/>
      <rosparam command="load" file="$(find adi_tmc_coe)/config/1slave_1axis.yaml"/>
    </node>
    
    <!--Launches tf frames-->
    <node pkg="tf" type="static_transform_publisher" name="$(eval arg('tmc_coe_base_name') + '_link')" 
      args="$(arg dev_pos_x) $(arg dev_pos_y) $(arg dev_pos_z) $(arg dev_pos_qx) $(arg dev_pos_qy) $(arg dev_pos_qz) $(arg dev_pos_qw) $(arg dev_parent_frame) $(arg dev_base_name) $(arg period)" />
 
    <node pkg="tf" type="static_transform_publisher" name="$(eval arg('tmc_coe_base_name') + '_mtr0_link')" 
      args="$(arg mtr0_pos_x) $(arg mtr0_pos_y) $(arg mtr0_pos_z) $(arg mtr0_pos_qx) $(arg mtr0_pos_qy) $(arg mtr0_pos_qz) $(arg mtr0_pos_qw) $(arg dev_base_name) $(arg mtr0_base_name) $(arg period)" />

  </group>

</launch>
